test/data/test_feature_store.py

```diff
@@ -0,0 +1,86 @@
+from typing import Optional
+
+import torch
+
+from torch_geometric.data.feature_store import FeatureStore, TensorAttr
+from torch_geometric.typing import FeatureTensorType
+
+
+class MyFeatureStore(FeatureStore):
+    r"""A basic feature store, does NOT implement all functionality of a
+    fully-fledged feature store. Only works for Torch tensors."""
+    def __init__(self):
+        super().__init__(backend='test')
+        self.store = {}
+
+    @classmethod
+    def key(cls, attr: TensorAttr):
+        return (attr.tensor_type or '', attr.node_type or '', attr.graph_type
+                or '')
+
+    def _put_tensor(self, tensor: FeatureTensorType, attr: TensorAttr) -> bool:
+        self.store[MyFeatureStore.key(attr)] = torch.cat(
+            (attr.index.reshape(-1, 1), tensor), dim=1)
+        return True
+
+    def _get_tensor(self, attr: TensorAttr) -> Optional[FeatureTensorType]:
+        tensor = self.store.get(MyFeatureStore.key(attr), None)
+        if tensor is None:
+            return None
+        if attr.index is not None:
```

Comment:
What does that do? What are we testing with this implementation?

Reply:
Made this cleaner :)